### DISCUSSION TITLE
PorousFlow - Pressure dependent sink

### DISCUSSION THREAD

**Discussion by MatiasAllay** [idaholab/moose#18327] [Q&A Modules: General] [answered]
Hi MOOSE,
I would like to implement a sink in which I set up a pressure condition and the flux in and out is computed internally. Just like `PorousFlowPiecewiseLinearSink` does. But I would like to implement it in the middle of my model rather than in a boundary. I looked at the sink documentation, but I couldn't find something that fit what I need.
Thanks in advance!
Matias

**Comment by WilkAndy** [comment, marked as answer]
If there is a boundary (eg, a 2D surface) in the middle of your model, then you should make one in the Mesh (there are mesh generators for that, eg SideSetsBetweenSubdomainsGenerator).

If there is no boundary, but instead you've got some point sources, then one of the PorousFlow DiracKernels should help: https://mooseframework.inl.gov/modules/porous_flow/sinks.html
